### PR TITLE
more support for lists

### PR DESCRIPTION
**Pull request by skimo-openhub**

### CODE REVIEW

**Comment by skimo-openhub** [comment]
https://github.com/facebookresearch/TensorComprehensions/tree/pr/list shows how this can be used in TC.

**Comment by skimo-openhub** [comment]
On Thu, Apr 05, 2018 at 05:18:48PM +0000, ftynse wrote:
> ftynse commented on this pull request.
> > @@ -2965,6 +2967,23 @@ It can be replaced using the following functions.
>  		__isl_take isl_multi_union_pw_aff *mupa, int pos,
>  		__isl_take isl_union_pw_aff *upa);
>  
> +A list of all base expression at of a multiple
> 
> "at of a multiple"

Thanks.  Fixed.

skimo

File: isl_union_map.c [hunk 1]
```diff
@@ -4165,6 +4165,44 @@ __isl_give isl_basic_set_list *isl_union_set_get_basic_set_list(
 	return list;
 }
 
+/* Add "set" to "list".
```
Comment (ftynse) [inline isl_union_map.c]: "list" is not a part of the signature here

File: isl_union_templ.c [hunk 2]
```diff
@@ -1170,3 +1170,39 @@ __isl_give UNION *FN(UNION,reset_user)(__isl_take UNION *u)
 	return FN(UNION,transform_space)(u, space, &FN(UNION,reset_user_entry),
 					NULL);
 }
+
+/* Add the base expression held by "entry" to "list".
```
Comment (ftynse) [inline isl_union_templ.c]: "list" is not a part of the signature here, it's wrapped in "user"

**Comment by skimo-openhub** [comment]
On Fri, Apr 06, 2018 at 04:53:02AM -0700, ftynse wrote:
> ftynse approved this pull request.
> 
> minor thing in comments to an internal function, LGTM otherwise
> 
> > @@ -4165,6 +4165,44 @@ __isl_give isl_basic_set_list *isl_union_set_get_basic_set_list(
>  	return list;
>  }
>  
> +/* Add "set" to "list".
> 
> "list" is not a part of the signature here
> 
> > @@ -1170,3 +1170,39 @@ __isl_give UNION *FN(UNION,reset_user)(__isl_take UNION *u)
>  	return FN(UNION,transform_space)(u, space, &FN(UNION,reset_user_entry),
>  					NULL);
>  }
> +
> +/* Add the base expression held by "entry" to "list".
> 
> "list" is not a part of the signature here, it's wrapped in "user"

That's right, but I think it's more clear this way
than referring to "user".
I may still change it if someone complains when I send it out
on the mailing list.

skimo

**Pull request merged**